udata/core/dataset/models.py

```diff
@@ -607,63 +607,65 @@ def quality(self):
             * description length
             * and so on
         """
-        from udata.models import Discussion  # noqa: Prevent circular imports
         result = {}
         if not self.id:
             # Quality is only relevant on saved Datasets
             return result
-        if self.frequency != 'unknown':
-            result['frequency'] = self.frequency
+
+        result['license'] = True if self.license else False
+        result['temporal_coverage'] = True if self.temporal_coverage else False
+        result['spatial'] = True if self.spatial else False
+
+        result['update_frequency'] = True if self.frequency != 'unknown' else False
         if self.next_update:
-            result['update_in'] = -(self.next_update - datetime.now()).days
-        if self.tags:
-            result['tags_count'] = len(self.tags)
-        if self.description:
-            result['description_length'] = len(self.description)
+            result['update_fulfilled_in_time'] = True if -(self.next_update - datetime.now()).days < 0 else False
+
+        result['tags_count'] = True if len(self.tags) > current_app.config.get('QUALITY_TAGS_COUNT') else False
+        result['dataset_description_quality'] = True if len(self.description) > current_app.config.get('QUALITY_DESCRIPTION_LENGTH') else False
+
         if self.resources:
             result['has_resources'] = True
             result['has_only_closed_or_no_formats'] = all(
```

Comment:
Could we rename this criterion to `has_open_formats`? I find double negation confusing